akka-typed/src/main/scala/akka/typed/patterns/Restarter.scala

```diff
@@ -7,6 +7,52 @@ package patterns
 import scala.reflect.ClassTag
```

Comment:
The patterns here are just Behaviors, i.e. not specific to a language DSL. The Restarter could just be moved into `akka.typed.internal`, but for the Receptionist users will need to use the message classes—either we move it to `akka.typed` or leave it here. But that does not necessarily need to be done in this PR.